src/main/java/com/arpnetworking/metrics/incubator/impl/TsdPeriodicMetricsFactory.java

```diff
@@ -0,0 +1,251 @@
+        final Metrics metrics = _currentPeriodicMetrics.getAndSetReference(_metricsFactory.create());
+        metrics.close();
+    }
+
+    private void recordPolledMetrics(final Metrics metrics) {
+        for (Consumer<Metrics> polledMetric : _polledMetricsRegistrations) {
+            polledMetric.accept(metrics);
+        }
+    }
+
+    private TsdPeriodicMetricsFactory(final Builder builder) {
+        _metricsFactory = builder._metricsFactory;
+
+        _currentPeriodicMetrics = new SafeRefLock<>(_metricsFactory.create());
+
+        _closingExecutor.scheduleAtFixedRate(this::cyclePeriodMetrics, 500, 500, TimeUnit.MILLISECONDS);
+    }
+
+    private final MetricsFactory _metricsFactory;
+    private final SafeRefLock<Metrics> _currentPeriodicMetrics;
+    private final ScheduledExecutorService _closingExecutor = new ScheduledThreadPoolExecutor(1, (r) -> new Thread(r, "MetricsCloser"));
+    private final Set<Consumer<Metrics>> _polledMetricsRegistrations = ConcurrentHashMap.newKeySet();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TsdPeriodicMetricsFactory.class);
+
+    /**
+     * Implementation of the Builder pattern for the {@link TsdPeriodicMetricsFactory} class.
+     *
+     * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+     */
+    public static final class Builder implements com.arpnetworking.commons.builder.Builder<TsdPeriodicMetricsFactory> {
+        /**
+         * Public constructor.
+         */
+        public Builder() {
+            this(LOGGER);
+        }
+
+        // NOTE: Package private for testing
+        /* package private */ Builder(@Nullable final Logger logger) {
+            _logger = logger;
+        }
+
+        /**
+         * Sets metrics factory. Required. Cannot be null.
+         *
+         * @param value The metrics factory.
+         * @return This instance of {@link Builder}.
+         */
+        public Builder setMetricsFactory(final MetricsFactory value) {
+            _metricsFactory = value;
+            return this;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public TsdPeriodicMetricsFactory build() {
+            final List<String> failures = new ArrayList<>();
+
+            // Defaults
+            if (_metricsFactory == null) {
+                _metricsFactory = new TsdMetricsFactory.Builder().build();
+                _logger.warn(String.format("Defaulted null metrics factory; metricsFactory=%s", _metricsFactory));
+            }
+
+            return new TsdPeriodicMetricsFactory(this);
+        }
+        private MetricsFactory _metricsFactory;
+
+        private final Logger _logger;
+    }
+
+    /* package private */ static final class NonClosingMetrics implements Metrics {
+        NonClosingMetrics(final Metrics wrapped) {
+            this._wrapped = wrapped;
+        }
+
+        @Override
+        public Counter createCounter(@Nonnull final String name) {
```

Comment:
You shouldn't need these if you include the the package-info.java file as in the client.

https://github.com/ArpNetworking/metrics-client-java/blob/master/src/main/java/com/arpnetworking/metrics/package-info.java